tool_path_planner/src/utilities.cpp

```diff
@@ -284,4 +299,115 @@ bool createToolPathSegment(const pcl::PointCloud<pcl::PointNormal>& cloud_normal
   return true;
 }
 
+// this is a helper function for splitPaths()
+double getDistance(Eigen::Isometry3d t1, Eigen::Isometry3d t2)
+{
+  auto p1 = t1.translation();
+  auto p2 = t2.translation();
+  return sqrt(pow(p1.x() - p2.x(), 2) + pow(p1.y() - p2.y(), 2) + pow(p1.z() - p2.z(), 2));
```

Comment:
You could do (p2 - p2).norm() I think

Reply:
This is an artifact born from the use of the keyword auto. When a programmer is not sure what kind of thing they are working with, they might assume it will take some extra work to dig out the data they want. I think we should remove this function, and also replace auto from nearly every for loop we have in our libraries. It helps readability and maintainability. auto should only be used when it is clear from nearby code exactly what the type is.